### PR TITLE
lsp: Implement bundle use in workspace Eval

### PR DESCRIPTION
**Pull request by charlieegan3**
This makes it possible to workspace eval rules based on data.

https://github.com/user-attachments/assets/0b79221b-21bc-4344-8873-a849222c9533

### CODE REVIEW

File: internal/lsp/bundles/cache.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	return refreshedBundles, nil
+}
+
+// List returns a list of all the bundle roots that are currently present in
+// the cache.
+func (c *Cache) List() []string {
+	roots := make([]string, 0, len(c.bundles))
+	for root := range c.bundles {
+		roots = append(roots, root)
+	}
```
Comment (anderseknert) [inline internal/lsp/bundles/cache.go:129]: Could use `util.Keys` :)

File: internal/lsp/bundles/cache.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+	// if there has been any change in any of the source files, then
+	// reload the bundle
+	c.bundle, err = LoadDataBundle(path)
+	if err != nil {
+		return false, fmt.Errorf("failed to load bundle %q: %w", path, err)
+	}
+
+	// update the bundle's sourceDigests to the new on-disk state after a
+	// successful refresh
+	c.sourceDigests = onDiskSourceDigests
```
Comment (anderseknert) [inline internal/lsp/bundles/cache.go:219]: Nice!

File: internal/lsp/server.go [hunk 3]
```diff
@@ -1887,6 +1896,17 @@ func (l *LanguageServer) loadWorkspaceContents(ctx context.Context, newOnly bool
...
 	}
 
+	if l.bundleCache != nil {
+		rb, err := l.bundleCache.Refresh()
+		if err != nil {
+			return nil, fmt.Errorf("failed to refresh the bundle cache: %w", err)
+		}
+
+		if len(rb) > 0 {
+			fmt.Fprintln(os.Stderr, "Reloaded bundles:", rb)
```
Comment (anderseknert) [inline internal/lsp/server.go:1906]: Do we need to print this each time?

**Review by anderseknert — APPROVED** [review]
Wow, awesome work!

File: internal/lsp/bundles/cache.go (hunk 1 above)
Comment (charlieegan3) [reply internal/lsp/bundles/cache.go]: Very nice, done in 9d2aa3bfad3159476acf2b8ba9ddf0e47cddde9a

File: internal/lsp/server.go (hunk 3 above)
Comment (charlieegan3) [reply internal/lsp/server.go]: I've dropped it in 9d2aa3bfad3159476acf2b8ba9ddf0e47cddde9a. It'd be nice to have some log level control but that can come another time.

**Pull request merged**